### PR TITLE
feat: Add metrics to track where signature rejection occurred

### PR DESCRIPTION
**Pull request by pnarayanaswamy**
Add a property location to the Signature Rejected event depending on where it was rejected from. 
location: `confirmation` or `alert_friction_modal`

## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26469?quickstart=1)

## **Related issues**

Fixes: [#2703](https://github.com/MetaMask/MetaMask-planning/issues/2703)

## **Manual testing steps**
 NOTE: All scenarios below have an E2E test in place along with this PR.

TC1: 
1. Go to the test dapp
2. Initiate any signature
3.  Reject it from the confirmation screen
4. Look for the Signature Rejected event in the network tab, the request should have the property location: "confirmation"

TC2:
1. Go to the test dapp
2. Initiate the signature 'SIGN IN WITH EHTEREUM (BAD DOMAIN)
3. Click on confirm
4.  Reject it from the alert
5. Look for the Signature Rejected event in the network tab, the request should have the property location: "alert_friction_modal"

## **Pre-merge author checklist**

- [x ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x ] I've completed the PR template to the best of my ability
- [x ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by pnarayanaswamy** [comment]
@SocketSecurity ignore-all

File: test/e2e/tests/confirmations/signatures/malicious-signatures.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  it('displays alert for domain binding and confirms', async function () {
+    await withRedesignConfirmationFixtures(
+      this.test?.fullTitle(),
+      async ({ driver }: TestSuiteArguments) => {
+        await unlockWallet(driver);
+        await openDapp(driver);
+        await driver.clickElement('#siweBadDomain');
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+
+        const alert = await driver.findElement('[data-testid="inline-alert"]');
```
Comment (matthewwalsh0) [inline test/e2e/tests/confirmations/signatures/malicious-signatures.spec.ts:23]: Pending the full page object model, could large parts of this go into reusable helpers such as `expectAlert` or `acknowledgeAlert` or `confirmWithAlertModal`?

File: ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx [hunk 2]
```diff
@@ -40,16 +41,20 @@ function ConfirmButtons({
   onSubmit,
   isConfirmed,
 }: {
-  onCancel: () => void;
+  onCancel: ({ location }: { location: MetaMetricsEventLocation }) => void;
```
Comment (matthewwalsh0) [inline ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx:44]: Minor, but as we define this a few times, is it worth an `OnCancelHandler` type exported from `footer.tsx`?

File: ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx [hunk 3]
```diff
@@ -40,16 +41,20 @@ function ConfirmButtons({
...
   onSubmit: () => void;
   isConfirmed: boolean;
 }) {
   const t = useI18nContext();
+  const onAlertCancel = useCallback(() => {
+    onCancel({ location: MetaMetricsEventLocation.AlertFrictionModal });
+  }, []);
```
Comment (matthewwalsh0) [inline ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx:51]: I think best practice is to include any properties as dependencies, so `onCancel` in this case?

File: ui/pages/confirmations/components/confirm/footer/footer.tsx [hunk 4]
```diff
@@ -153,12 +157,16 @@ const Footer = () => {
     }
   }, [currentConfirmation, customNonceValue]);
 
+  const onFooterCancel = useCallback(() => {
+    onCancel({ location: MetaMetricsEventLocation.Confirmation });
+  }, [currentConfirmation]);
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/footer/footer.tsx:162]: Also here, I think we should include `onCancel` as a dependency.

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: shared/constants/metametrics.ts [hunk 5]
```diff
@@ -820,6 +820,8 @@ export enum MetaMetricsEventLocation {
   TokenDetection = 'token_detection',
   TokenMenu = 'token_menu',
   Transaction = 'transaction',
+  Confirmation = 'confirmation',
+  AlertFrictionModal = 'alert_friction_modal',
```
[the remark this answers is not in the document]
Comment (pnarayanaswamy) [reply shared/constants/metametrics.ts]: done

File: test/e2e/tests/confirmations/signatures/malicious-signatures.spec.ts (hunk 1 above)
Comment (pnarayanaswamy) [reply test/e2e/tests/confirmations/signatures/malicious-signatures.spec.ts]: done

File: ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx (hunk 2 above)
Comment (pnarayanaswamy) [reply ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx]: done

File: ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx (hunk 3 above)
Comment (pnarayanaswamy) [reply ui/components/app/alert-system/confirm-alert-modal/confirm-alert-modal.tsx]: done

File: ui/pages/confirmations/components/confirm/footer/footer.tsx (hunk 4 above)
Comment (pnarayanaswamy) [reply ui/pages/confirmations/components/confirm/footer/footer.tsx]: done

File: ui/pages/confirmations/components/confirm/footer/footer.tsx [hunk 6]
```diff
@@ -29,6 +29,12 @@ import { REDESIGN_TRANSACTION_TYPES } from '../../../utils';
 import { getConfirmationSender } from '../utils';
 import { MetaMetricsEventLocation } from '../../../../../../shared/constants/metametrics';
 
+export type OnCancelType = ({
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/footer/footer.tsx:32]: Minor, this is exactly what I meant but the `Type` is implied, so maybe `OnCancelHandler` or `OnCancelCallback`?

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/footer/footer.tsx (hunk 6 above)
Comment (pnarayanaswamy) [reply ui/pages/confirmations/components/confirm/footer/footer.tsx]: done, renamed

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Pull request merged**